Tensile/TensileCreateLibrary.py

```diff
@@ -396,69 +397,78 @@ def buildSourceCodeObjectFiles(CxxCompiler, kernelFiles, outputPath):
 
 
 ################################################################################
-def prepAsm(kernelWriterAssembly):
-    """
-    Create and prepare the assembly directory  - called ONCE per output dir:
+def prepAsm(kernelWriterAssembly: KernelWriterAssembly, buildPath: Path):
+    """Create and prepare the assembly directory. - called ONCE per output directory.
+
+    This function is called once per output directory. It creates a directory
+    "assembly" under the provided **buildPath**, and generates a bash script for
+    compiling object files into code object files.
+
+    Args:
+        kernelWriterAssembly: Assembly writer object.
+        buildPath: Path to directory where assembly files will be written.
     """
-    asmPath = ensurePath(os.path.join(globalParameters["WorkingPath"], "assembly"))
+    asmPath = buildPath / "assembly"
+    asmPath.mkdir(exist_ok=True)
+
     isa = globalParameters["CurrentISA"]
```

Comment:
What about this global?

Reply:
Good catch, was focused on the "WorkingPath" parameter and had blinders on.